### PR TITLE
Feat/complete pnc auth

### LINKED ISSUE #54: Handle rejection of authorization in 15118-2
Correctly handle the case where the AuthorizationReq is rejected, in the PaymentDetails and Authorization states.  This should also include handling the case where the authorization is Ongoing (returns early to comply with ISO 15118 timeouts, but is continuing to be processed, and will be requested again).

Per @shalinnijel2 :

If Authorization is rejected we need to call:

            self.stop_state_machine(
                "Authorization was rejected",
                message,
                ResponseCode.FAILED,
            )

'message' above is the incoming AuthorizationReq message. stop_state_machine() would build the appropriate response message - in this case AuthorizationRes message with ResponseCode set to ResponseCode.FAILED and also set the next_state to Terminate.
One thing to keep in mind here is the failed response code in AuthorizationRes could be FAILED or as per V2G2-475, be FAILED_ChallengeInvalid - but I think that will be better handled under a separate ticket

Per @tropxy :

in case of Ongoing or Accepted we go to the Authorization state.
in case of rejected then we can answer with an Error code.

we also need to save the state of the authorization in the session here, I believe, because if the auth is successful in the PaymentDetailsReq, then we wont need to request it in the Authorization state and we can move forward. Unless, the ocpp client holds the authorization response and answers immediately if we do another request for the same eMAID.

if the answer "accepted" was not yet received, then we will keep requesting until the timeout (60s)

### PR DESCRIPTION
**Pull request by tropxy**
Fix #54

### CODE REVIEW

File: iso15118/secc/controller/simulator.py [hunk 1]
```diff
@@ -423,12 +423,12 @@ async def get_energy_service_list(self) -> ServiceList:
...
         self,
         id_token: Optional[str] = None,
-        id_token_type: Optional[AuthEnum] = None,
+        id_token_type: Optional[AuthorizationTokenType] = None,
         certificate_chain: Optional[bytes] = None,
         hash_data: Optional[List[Dict[str, str]]] = None,
     ) -> AuthorizationStatus:
         """Overrides EVSEControllerInterface.is_authorized()."""
-        return AuthorizationStatus.ACCEPTED
+        return AuthorizationStatus.ONGOING
```
Comment (shalinnijel2) [inline iso15118/secc/controller/simulator.py:431]: Should this be set back to 
```
return AuthorizationStatus.ACCEPTED
```

**Review by shalinnijel2 — CHANGES_REQUESTED** [review]
One minor change...rest lgtm...

File: iso15118/secc/controller/simulator.py (hunk 1 above)
Comment (tropxy) [reply iso15118/secc/controller/simulator.py:431]: yes, thanks ;)

Comment (tropxy) [reply iso15118/secc/controller/simulator.py]: done

**Review by shalinnijel2 — APPROVED** [review]

**Pull request merged**